[JSC] Add simple ByVal handlers

Constellation opened the pull request:
#### 8b9a484586ea691225d52739ef780a8878ed8fbd
<pre>
[JSC] Add simple ByVal handlers
<a href="https://bugs.webkit.org/show_bug.cgi?id=275644">https://bugs.webkit.org/show_bug.cgi?id=275644</a>
<a href="https://rdar.apple.com/130117255">rdar://130117255</a>

Reviewed by Yijia Huang.

This patch adds simple handlers for ByVal cases. In particular, this patch adds GetByVal Load/Miss, PutByVal Replace/Transition, and InByVal InHit/InMiss.
These cases are dominated in code-generating handlers right now.

* Source/JavaScriptCore/bytecode/InlineCacheCompiler.cpp:
(JSC::InlineCacheCompiler::emitDataICCheckStructure):
(JSC::InlineCacheCompiler::emitDataICCheckUid):
(JSC::InlineCacheHandler::createPreCompiled):
(JSC::loadHandlerImpl):
(JSC::getByIdLoadHandlerCodeGeneratorImpl):
(JSC::getByIdMissHandlerCodeGenerator):
(JSC::getByIdCustomHandlerImpl):
(JSC::getByIdGetterHandler):
(JSC::getByIdProxyObjectLoadHandler):
(JSC::putByIdReplaceHandlerCodeGenerator):
(JSC::transitionHandlerImpl):
(JSC::putByIdTransitionHandlerCodeGeneratorImpl):
(JSC::putByIdCustomHandlerImpl):
(JSC::putByIdSetterHandlerImpl):
(JSC::inByIdInHandlerImpl):
(JSC::instanceOfHandlerImpl):
(JSC::getByValLoadHandlerCodeGeneratorImpl):
(JSC::getByValWithStringLoadOwnPropertyHandlerCodeGenerator):
(JSC::getByValWithStringLoadPrototypePropertyHandlerCodeGenerator):
(JSC::getByValWithSymbolLoadOwnPropertyHandlerCodeGenerator):
(JSC::getByValWithSymbolLoadPrototypePropertyHandlerCodeGenerator):
(JSC::getByValMissHandlerCodeGeneratorImpl):
(JSC::getByValWithStringMissHandlerCodeGenerator):
(JSC::getByValWithSymbolMissHandlerCodeGenerator):
(JSC::putByValReplaceHandlerCodeGeneratorImpl):
(JSC::putByValWithStringReplaceHandlerCodeGenerator):
(JSC::putByValWithSymbolReplaceHandlerCodeGenerator):
(JSC::putByValTransitionHandlerCodeGeneratorImpl):
(JSC::putByValWithStringTransitionNonAllocatingHandlerCodeGenerator):
(JSC::putByValWithSymbolTransitionNonAllocatingHandlerCodeGenerator):
(JSC::putByValWithStringTransitionNewlyAllocatingHandlerCodeGenerator):
(JSC::putByValWithSymbolTransitionNewlyAllocatingHandlerCodeGenerator):
(JSC::putByValWithStringTransitionReallocatingHandlerCodeGenerator):
(JSC::putByValWithSymbolTransitionReallocatingHandlerCodeGenerator):
(JSC::inByValInHandlerImpl):
(JSC::inByValWithStringHitHandler):
(JSC::inByValWithStringMissHandler):
(JSC::inByValWithSymbolHitHandler):
(JSC::inByValWithSymbolMissHandler):
(JSC::InlineCacheCompiler::compileOneAccessCaseHandler):
(JSC::InlineCacheCompiler::compileGetByIdDOMJITHandler):
* Source/JavaScriptCore/bytecode/InlineCacheCompiler.h:
* Source/JavaScriptCore/jit/GPRInfo.h:
(JSC::JSValueRegs::payloadOnly):
(JSC::JSValueRegs::withTwoAvailableRegs):
(JSC::JSValueRegs::operator! const):
(JSC::JSValueRegs::operator bool const):
* Source/JavaScriptCore/jit/JITThunks.h:

Canonical link: <a href="https://commits.webkit.org/280199@main">https://commits.webkit.org/280199@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/2c3934ff0abed059f59731db7625a1829852c2cb

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/55944 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/35268 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/8412 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/58943 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/6376 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/42889 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/6572 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/45052 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/4407 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/57973 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/33171 "Found 60 new test failures: accessibility/accessibility-node-reparent.html accessibility/ancestor-computation.html accessibility/announcement-notification.html accessibility/aria-braillelabel.html accessibility/aria-brailleroledescription.html accessibility/aria-checked-mixed-value.html accessibility/aria-current-state-changed-notification.html accessibility/aria-current.html accessibility/aria-describedby-on-input.html accessibility/aria-description.html ... (failure)") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/48236 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/26190 "Passed tests") | 
| | [❌ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/29954 "Found 60 new test failures: imported/w3c/web-platform-tests/FileAPI/Blob-methods-from-detached-frame.html imported/w3c/web-platform-tests/FileAPI/FileReader/Progress_event_bubbles_cancelable.html imported/w3c/web-platform-tests/FileAPI/FileReader/workers.html imported/w3c/web-platform-tests/FileAPI/FileReaderSync.worker.html imported/w3c/web-platform-tests/FileAPI/blob/Blob-array-buffer.any.html imported/w3c/web-platform-tests/FileAPI/blob/Blob-array-buffer.any.worker.html imported/w3c/web-platform-tests/FileAPI/blob/Blob-bytes.any.html imported/w3c/web-platform-tests/FileAPI/blob/Blob-bytes.any.worker.html imported/w3c/web-platform-tests/FileAPI/blob/Blob-constructor-dom.window.html imported/w3c/web-platform-tests/FileAPI/blob/Blob-constructor-endings.html ... (failure)") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/5564 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/4519 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/49020 "Built successfully and passed tests") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/51926 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/5835 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/60530 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/55180 "Built successfully and passed tests") | [⏳ 🛠 vision ](https://ews-build.webkit.org/#/builders/visionOS-1-Build-EWS "Waiting in queue, processing has not started yet") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/5956 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/52480 "Passed tests") | 
| | [⏳ 🛠 vision-sim ](https://ews-build.webkit.org/#/builders/visionOS-1-Simulator-Build-EWS "Waiting in queue, processing has not started yet") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/48305 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/51999 "Passed tests") | 
| | [⏳ 🧪 vision-wk2 ](https://ews-build.webkit.org/#/builders/visionOS-1-Simulator-WK2-Tests-EWS "Waiting in queue, processing has not started yet") | | [  ~~🛠 jsc-armv7~~](https://ews-build.webkit.org/#/builders/35/builds/76941 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/8278 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/31097 "Built successfully") | | [  ~~🧪 jsc-armv7-tests~~](https://ews-build.webkit.org/#/builders/35/builds/76941 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/32181 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/33263 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/31929 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

hyjorc1 commented on Source/JavaScriptCore/bytecode/InlineCacheCompiler.h:
```diff
@@ -435,6 +437,20 @@ MacroAssemblerCodeRef<JITThunkPtrTag> inByIdHitHandler(VM&);
 MacroAssemblerCodeRef<JITThunkPtrTag> inByIdMissHandler(VM&);
 MacroAssemblerCodeRef<JITThunkPtrTag> instanceOfHitHandler(VM&);
 MacroAssemblerCodeRef<JITThunkPtrTag> instanceOfMissHandler(VM&);
+MacroAssemblerCodeRef<JITThunkPtrTag> getByValWithStringLoadOwnPropertyHandlerCodeGenerator(VM&);
```
I think this handler is responsible for loading a property from an object when the property key is a string, and the property is directly owned by the object.
```
obj["propertyName"]
```
When a JavaScript code accesses a property using a string key directly on an object, this handler validates the obj's structure and property's key (a string in this case) and then directly accesses the property value from the cached location.

hyjorc1 commented on Source/JavaScriptCore/bytecode/InlineCacheCompiler.h:
```diff
@@ -435,6 +437,24 @@ MacroAssemblerCodeRef<JITThunkPtrTag> inByIdHitHandler(VM&);
 MacroAssemblerCodeRef<JITThunkPtrTag> instanceOfHitHandler(VM&);
 MacroAssemblerCodeRef<JITThunkPtrTag> instanceOfMissHandler(VM&);
+MacroAssemblerCodeRef<JITThunkPtrTag> getByValWithStringLoadOwnPropertyHandlerCodeGenerator(VM&);
+MacroAssemblerCodeRef<JITThunkPtrTag> getByValWithStringLoadPrototypePropertyHandlerCodeGenerator(VM&);
```
Similarly, this handler deals with loading a property when the property key is a string, and the property is found on the prototype chain, not directly on the object.
```
obj["inheritedProperty"]
```

hyjorc1 commented on Source/JavaScriptCore/bytecode/InlineCacheCompiler.h:
```diff
@@ -435,6 +437,24 @@ MacroAssemblerCodeRef<JITThunkPtrTag> inByIdHitHandler(VM&);
 MacroAssemblerCodeRef<JITThunkPtrTag> instanceOfMissHandler(VM&);
+MacroAssemblerCodeRef<JITThunkPtrTag> getByValWithStringLoadOwnPropertyHandlerCodeGenerator(VM&);
+MacroAssemblerCodeRef<JITThunkPtrTag> getByValWithStringLoadPrototypePropertyHandlerCodeGenerator(VM&);
+MacroAssemblerCodeRef<JITThunkPtrTag> getByValWithStringMissHandlerCodeGenerator(VM&);
```
This handler is used when a property with a string key is not found on the object or its prototype chain.
```
obj["nonExistentProperty"]
```

hyjorc1 commented on Source/JavaScriptCore/bytecode/InlineCacheCompiler.h:
```diff
@@ -435,6 +437,24 @@ MacroAssemblerCodeRef<JITThunkPtrTag> inByIdHitHandler(VM&);
+MacroAssemblerCodeRef<JITThunkPtrTag> getByValWithStringLoadOwnPropertyHandlerCodeGenerator(VM&);
+MacroAssemblerCodeRef<JITThunkPtrTag> getByValWithStringLoadPrototypePropertyHandlerCodeGenerator(VM&);
+MacroAssemblerCodeRef<JITThunkPtrTag> getByValWithStringMissHandlerCodeGenerator(VM&);
+MacroAssemblerCodeRef<JITThunkPtrTag> getByValWithSymbolLoadOwnPropertyHandlerCodeGenerator(VM&);
```
This handler is for loading a property from an object using a symbol key, where the property is owned directly by the object.
```
obj[someSymbol]
```

hyjorc1 commented on Source/JavaScriptCore/bytecode/InlineCacheCompiler.h:
```diff
@@ -435,6 +437,24 @@ MacroAssemblerCodeRef<JITThunkPtrTag> inByIdHitHandler(VM&);
+MacroAssemblerCodeRef<JITThunkPtrTag> getByValWithStringLoadPrototypePropertyHandlerCodeGenerator(VM&);
+MacroAssemblerCodeRef<JITThunkPtrTag> getByValWithStringMissHandlerCodeGenerator(VM&);
+MacroAssemblerCodeRef<JITThunkPtrTag> getByValWithSymbolLoadOwnPropertyHandlerCodeGenerator(VM&);
+MacroAssemblerCodeRef<JITThunkPtrTag> getByValWithSymbolLoadPrototypePropertyHandlerCodeGenerator(VM&);
```
This handler loads a property when the key is a symbol and the property is found in the prototype.
```
obj[inheritedSymbol] 
```

hyjorc1 commented on Source/JavaScriptCore/bytecode/InlineCacheCompiler.h:
```diff
@@ -435,6 +437,24 @@ MacroAssemblerCodeRef<JITThunkPtrTag> inByIdHitHandler(VM&);
+MacroAssemblerCodeRef<JITThunkPtrTag> getByValWithStringMissHandlerCodeGenerator(VM&);
+MacroAssemblerCodeRef<JITThunkPtrTag> getByValWithSymbolLoadOwnPropertyHandlerCodeGenerator(VM&);
+MacroAssemblerCodeRef<JITThunkPtrTag> getByValWithSymbolLoadPrototypePropertyHandlerCodeGenerator(VM&);
+MacroAssemblerCodeRef<JITThunkPtrTag> getByValWithSymbolMissHandlerCodeGenerator(VM&);
```
For the "miss" case in property accesses where the key is a symbol.

hyjorc1 commented on Source/JavaScriptCore/bytecode/InlineCacheCompiler.h:
```diff
@@ -435,6 +437,24 @@ MacroAssemblerCodeRef<JITThunkPtrTag> inByIdHitHandler(VM&);
+MacroAssemblerCodeRef<JITThunkPtrTag> getByValWithSymbolLoadOwnPropertyHandlerCodeGenerator(VM&);
+MacroAssemblerCodeRef<JITThunkPtrTag> getByValWithSymbolLoadPrototypePropertyHandlerCodeGenerator(VM&);
+MacroAssemblerCodeRef<JITThunkPtrTag> getByValWithSymbolMissHandlerCodeGenerator(VM&);
+MacroAssemblerCodeRef<JITThunkPtrTag> putByValWithStringReplaceHandlerCodeGenerator(VM&);
```
This handler deals with replacing an existing property on an object with a new value, where the key is a string.
```
obj["propertyName"] = newValue
```

hyjorc1 commented on Source/JavaScriptCore/bytecode/InlineCacheCompiler.h:
```diff
@@ -435,6 +437,24 @@ MacroAssemblerCodeRef<JITThunkPtrTag> inByIdHitHandler(VM&);
+MacroAssemblerCodeRef<JITThunkPtrTag> getByValWithSymbolLoadPrototypePropertyHandlerCodeGenerator(VM&);
+MacroAssemblerCodeRef<JITThunkPtrTag> getByValWithSymbolMissHandlerCodeGenerator(VM&);
+MacroAssemblerCodeRef<JITThunkPtrTag> putByValWithStringReplaceHandlerCodeGenerator(VM&);
+MacroAssemblerCodeRef<JITThunkPtrTag> putByValWithStringTransitionNonAllocatingHandlerCodeGenerator(VM&);
```
Handles property assignment with a string key where the operation doesn't require memory allocation (non-allocating transition) for the new property.
```
obj["newProperty"] = value
```

hyjorc1 commented on Source/JavaScriptCore/bytecode/InlineCacheCompiler.h:
```diff
@@ -435,6 +437,24 @@ MacroAssemblerCodeRef<JITThunkPtrTag> inByIdHitHandler(VM&);
+MacroAssemblerCodeRef<JITThunkPtrTag> getByValWithSymbolMissHandlerCodeGenerator(VM&);
+MacroAssemblerCodeRef<JITThunkPtrTag> putByValWithStringReplaceHandlerCodeGenerator(VM&);
+MacroAssemblerCodeRef<JITThunkPtrTag> putByValWithStringTransitionNonAllocatingHandlerCodeGenerator(VM&);
+MacroAssemblerCodeRef<JITThunkPtrTag> putByValWithStringTransitionNewlyAllocatingHandlerCodeGenerator(VM&);
```
Manages assignments where the property key is a string, and adding the property requires allocation of new space (newly allocating transition).
```
obj["anotherNewProperty"] = value
```

hyjorc1 commented on Source/JavaScriptCore/bytecode/InlineCacheCompiler.h:
```diff
@@ -435,6 +437,24 @@ MacroAssemblerCodeRef<JITThunkPtrTag> inByIdHitHandler(VM&);
+MacroAssemblerCodeRef<JITThunkPtrTag> putByValWithStringReplaceHandlerCodeGenerator(VM&);
+MacroAssemblerCodeRef<JITThunkPtrTag> putByValWithStringTransitionNonAllocatingHandlerCodeGenerator(VM&);
+MacroAssemblerCodeRef<JITThunkPtrTag> putByValWithStringTransitionNewlyAllocatingHandlerCodeGenerator(VM&);
+MacroAssemblerCodeRef<JITThunkPtrTag> putByValWithStringTransitionReallocatingHandlerCodeGenerator(VM&);
```
Handles property assignments where the key is a string, and reallocation of existing space is necessary (reallocating transition).

My understanding for `Non-Allocating`, `Newly Allocating`, and `Reallocating ` are:

Non-Allocating: Uses existing inline storage without requiring additional memory allocation.

Newly Allocating: Allocates new memory to accommodate additional properties but doesn’t involve relocating existing storage.

Reallocating: Involves both allocating new memory and copying existing properties to new, larger storage.

hyjorc1 commented on Source/JavaScriptCore/bytecode/InlineCacheCompiler.h:
```diff
@@ -435,6 +437,24 @@ MacroAssemblerCodeRef<JITThunkPtrTag> inByIdHitHandler(VM&);
+MacroAssemblerCodeRef<JITThunkPtrTag> putByValWithStringTransitionNonAllocatingHandlerCodeGenerator(VM&);
+MacroAssemblerCodeRef<JITThunkPtrTag> putByValWithStringTransitionNewlyAllocatingHandlerCodeGenerator(VM&);
+MacroAssemblerCodeRef<JITThunkPtrTag> putByValWithStringTransitionReallocatingHandlerCodeGenerator(VM&);
+MacroAssemblerCodeRef<JITThunkPtrTag> putByValWithSymbolReplaceHandlerCodeGenerator(VM&);
+MacroAssemblerCodeRef<JITThunkPtrTag> putByValWithSymbolTransitionNonAllocatingHandlerCodeGenerator(VM&);
+MacroAssemblerCodeRef<JITThunkPtrTag> putByValWithSymbolTransitionNewlyAllocatingHandlerCodeGenerator(VM&);
+MacroAssemblerCodeRef<JITThunkPtrTag> putByValWithSymbolTransitionReallocatingHandlerCodeGenerator(VM&);
```
Similar to the string replace handler but for properties with symbol keys.

hyjorc1 reviewed (approved):
r=me

Constellation committed:
[JSC] Add simple ByVal handlers
https://bugs.webkit.org/show_bug.cgi?id=275644
rdar://130117255

Reviewed by Yijia Huang.

This patch adds simple handlers for ByVal cases. In particular, this patch adds GetByVal Load/Miss, PutByVal Replace/Transition, and InByVal InHit/InMiss.
These cases are dominated in code-generating handlers right now.

* Source/JavaScriptCore/bytecode/InlineCacheCompiler.cpp:
(JSC::InlineCacheCompiler::emitDataICCheckStructure):
(JSC::InlineCacheCompiler::emitDataICCheckUid):
(JSC::InlineCacheHandler::createPreCompiled):
(JSC::loadHandlerImpl):
(JSC::getByIdLoadHandlerCodeGeneratorImpl):
(JSC::getByIdMissHandlerCodeGenerator):
(JSC::getByIdCustomHandlerImpl):
(JSC::getByIdGetterHandler):
(JSC::getByIdProxyObjectLoadHandler):
(JSC::putByIdReplaceHandlerCodeGenerator):
(JSC::transitionHandlerImpl):
(JSC::putByIdTransitionHandlerCodeGeneratorImpl):
(JSC::putByIdCustomHandlerImpl):
(JSC::putByIdSetterHandlerImpl):
(JSC::inByIdInHandlerImpl):
(JSC::instanceOfHandlerImpl):
(JSC::getByValLoadHandlerCodeGeneratorImpl):
(JSC::getByValWithStringLoadOwnPropertyHandlerCodeGenerator):
(JSC::getByValWithStringLoadPrototypePropertyHandlerCodeGenerator):
(JSC::getByValWithSymbolLoadOwnPropertyHandlerCodeGenerator):
(JSC::getByValWithSymbolLoadPrototypePropertyHandlerCodeGenerator):
(JSC::getByValMissHandlerCodeGeneratorImpl):
(JSC::getByValWithStringMissHandlerCodeGenerator):
(JSC::getByValWithSymbolMissHandlerCodeGenerator):
(JSC::putByValReplaceHandlerCodeGeneratorImpl):
(JSC::putByValWithStringReplaceHandlerCodeGenerator):
(JSC::putByValWithSymbolReplaceHandlerCodeGenerator):
(JSC::putByValTransitionHandlerCodeGeneratorImpl):
(JSC::putByValWithStringTransitionNonAllocatingHandlerCodeGenerator):
(JSC::putByValWithSymbolTransitionNonAllocatingHandlerCodeGenerator):
(JSC::putByValWithStringTransitionNewlyAllocatingHandlerCodeGenerator):
(JSC::putByValWithSymbolTransitionNewlyAllocatingHandlerCodeGenerator):
(JSC::putByValWithStringTransitionReallocatingHandlerCodeGenerator):
(JSC::putByValWithSymbolTransitionReallocatingHandlerCodeGenerator):
(JSC::inByValInHandlerImpl):
(JSC::inByValWithStringHitHandler):
(JSC::inByValWithStringMissHandler):
(JSC::inByValWithSymbolHitHandler):
(JSC::inByValWithSymbolMissHandler):
(JSC::InlineCacheCompiler::compileOneAccessCaseHandler):
(JSC::InlineCacheCompiler::compileGetByIdDOMJITHandler):
* Source/JavaScriptCore/bytecode/InlineCacheCompiler.h:
* Source/JavaScriptCore/jit/GPRInfo.h:
(JSC::JSValueRegs::payloadOnly):
(JSC::JSValueRegs::withTwoAvailableRegs):
(JSC::JSValueRegs::operator! const):
(JSC::JSValueRegs::operator bool const):
* Source/JavaScriptCore/jit/JITThunks.h:

Canonical link: https://commits.webkit.org/280199@main
